internal/template/unstructured.go

```diff
@@ -83,3 +84,31 @@ func applySliceFieldFunc(obj map[string]any, path []string, ff fieldFunc) error
 	}
 	return nil
 }
+
+func processNestedSlice(obj map[string]any, path []string, ff fieldFunc) error {
+	ind := slices.Index(path, "[]")
+	if ind == len(path)-1 {
+		return applySliceFieldFunc(obj, path[:ind], ff)
+	}
```

Comment:
This code could be cleaned up a bit. It currently mixes two coding levels - a higher level call to a function that handles a particular case: applySliceFieldFunc - and lower level code that deals directly with the other case.

I think the code below this line aught to be folded into its own function.

Reply:
+1